p2p/src/network/pubsub/p2p_network_pubsub_reducer.rs

```diff
@@ -440,6 +437,126 @@ impl P2pNetworkPubsubState {
                 }
                 Ok(())
             }
+            P2pNetworkPubsubAction::BroadcastValidationCallback {
+                message,
+                message_content,
+                peer_id,
+                result,
+            } => {
+                let message_id = pubsub_state.mcache.put(message.clone());
```

Comment:
As mentioned in the other thread, here we are adding messages to the cache even in cases where we know we don't need them (rejected or ignored)